src/constants/constants.h

```diff
@@ -8,13 +8,21 @@
 
 namespace subbots {
 namespace global_constants {
-    // minimum height in cm from the camera's position that an
-    // overhead obstacle should be
-    static const float CLEARANCE_HEIGHT = 60;
+    static const float TARGET_TOP_POLE_CLEARANCE = 2.0;
 
-    // minimum width in cm from camera's position that
-    // side obstacles should be
-    static const float CLEARANCE_WIDTH = 30;
+    static const float ERROR_TOLERANCE_TOP_POLE_CLEARANCE = 0.1;
+
+    static const float TARGET_TOP_POLE_DISTANCE = 6;
+
+    static const float ERROR_TOLERANCE_TOP_POLE_DISTANCE = 0.05;
+
+    static const float ERROR_TOLERANCE_SIDE_POLES_ANGLE = 0.1;
+
+    static const float TARGET_SIDE_POLES_DISTANCE = 6;
+
+    static const float ERROR_TOLERANCE_SIDE_POLES_DISTANCE = 0.05;
+
+    static const float TARGET_AVERAGE_GATE_DISTANCE = 5;
```

Comment:
These seem like they should all be rosparams, as we'll likely have to change all of them a bunch as we start testing, and it will be a lot faster if we don't have to rebuild each time.

Reply:
Yeah I definitely agree, but they need to be accessed by both worldstate and decision, what's the best way to go about this?

Reply:
Because the timeline is so compressed I'm just going to say leave it for now; we can come back and improve this later if there's time. Just so you know, a potential option would be to use the `<arg>` tag in a roslaunch file, ex:
```
<launch>

    <arg name="error_tolerance_side_poles_distance" value="0.05"/>

    <node name="decision" pkg="decision" type="decision">
        <param name="error_tolerance_side_poles_distance" value="$(arg error_tolerance_side_poles_distance)" />
    </node>

    <node name="worldstate" pkg="worldstate" type="worldstate">
        <param name="error_tolerance_side_poles_distance" value="$(arg error_tolerance_side_poles_distance)" />
    </node>

</launch>
```
you could also put this configuration options in a `.yaml` file that the `roslaunch` file imports.
ie in `constants.yaml`:
```
error_tolerance_side_poles_distance: 0.05
target_average_gate distance: 0.05
```
and in the roslaunch file:
```
<launch>


    <node name="decision" pkg="decision" type="decision">
        <rosparam command="load" file="$(find constants)/constants.yaml" />
    </node>

    <node name="worldstate" pkg="worldstate" type="worldstate">
        <rosparam command="load" file="$(find constants)/constants.yaml" />
    </node>

</launch>
```

Reply:
(Note: I didn't test anything above, it's just to give a general idea of what it would look like)

Reply:
Oh, just saw your comments below. I'd say go for it if you're already working on this!